### PR TITLE
Bring closer to golang community style

### PR DESCRIPTION
**Pull request by philoserf**
`gofmt -s -w -l .&&goreturns -b -i -w -l` is my goto final check for golang projects.

- `gofmt` applies community style to the code
- `gofmt` with the `-s` flag simplifies code where it can
- `goreturns` with the `-i` flag runs `goimports` which uses community style for imports ordering and other goodness
- remove lint (`gometalinter --vendored-linters --enable-all --vendor main.go`) except gocyclo,
- follow community style for function and type names when we do not need to export them

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -35,100 +33,93 @@ import (
...
 }
 
-// Contains State that are read in from the command
-// line when the program is invoked.
-type State struct {
-	Client         *http.Client
-	Cookies        string
-	Expanded       bool
+// state contains data read in from the command line when the program is invoked
+type state struct {
```
Comment (philoserf) [inline main.go]: structalign: align the struct for better memory allocation

File: main.go [hunk 2]
```diff
@@ -35,100 +33,93 @@ import (
...
+
+const (
+	dirMode = "dir"
+	dnsMode = "dns"
 )
 
-// A single result which comes from an individual web
-// request.
-type Result struct {
+// result is the response of an individual web request
```
Comment (philoserf) [inline main.go]: standard format for a doc string is `// Name ...`

File: main.go [hunk 3]
```diff
@@ -35,100 +33,93 @@ import (
...
+type result struct {
```
Comment (philoserf) [inline main.go]: since nothing here is exported for other packages to use we can use lower case names

File: main.go [hunk 4]
```diff
@@ -197,16 +188,20 @@ func MakeRequest(s *State, fullUrl, cookie string) (*int, *int64) {
...
-			if re, ok := ue.Err.(*RedirectError); ok {
+			if re, ok := ue.Err.(*redirectError); ok {
 				return &re.StatusCode, nil
 			}
 		}
 		return nil, nil
 	}
 
-	defer resp.Body.Close()
+	defer func() {
```
Comment (philoserf) [inline main.go]: check the error of the deferred function call

File: main.go [hunk 5]
```diff
@@ -197,16 +188,20 @@ func MakeRequest(s *State, fullUrl, cookie string) (*int, *int64) {
...
+		if err := resp.Body.Close(); err != nil {
+			log.Print(err)
+		}
+	}()
 
-	var length *int64 = nil
+	var length *int64
```
Comment (philoserf) [inline main.go]: use golang's zero value

File: main.go [hunk 6]
```diff
@@ -361,15 +354,15 @@ func ParseCmdLine() *State {
...
+					s.StatusCodes.add(i)
 				}
 			}
 		}
 
 		// prompt for password if needed
 		if valid && s.Username != "" && s.Password == "" {
 			fmt.Printf("[?] Auth Password: ")
-			passBytes, err := terminal.ReadPassword(int(syscall.Stdin))
+			passBytes, err := terminal.ReadPassword(syscall.Stdin)
```
Comment (philoserf) [inline main.go]: no need to cast

File: main.go [hunk 7]
```diff
@@ -384,36 +377,35 @@ func ParseCmdLine() *State {
 		}
 
 		if valid {
-			var proxyUrlFunc func(*http.Request) (*url.URL, error)
-			proxyUrlFunc = http.ProxyFromEnvironment
+			proxyURL := http.ProxyFromEnvironment
 
 			if proxy != "" {
-				proxyUrl, err := url.Parse(proxy)
+				p, err := url.Parse(proxy)
```
Comment (philoserf) [inline main.go]: there were two things with very similar names
so here i use a short name for the most local variable

File: main.go [hunk 8]
```diff
@@ -424,32 +416,29 @@ func ParseCmdLine() *State {
...
-func Process(s *State) {
+// process the busting of the website with the given set of settings from the command line.
+func process(s *state) {
 
-	ShowConfig(s)
+	showConfig(s)
 
-	if s.Setup(s) == false {
-		Ruler(s)
+	if !s.Setup(s) {
```
Comment (philoserf) [inline main.go]: flip the boolean check for brevity since the return value is a boolean

File: main.go [hunk 9]
```diff
@@ -482,23 +469,25 @@ func Process(s *State) {
...
+		}()
 
-		// Lazy reading of the wordlist line by line
 		scanner = bufio.NewScanner(wordlist)
 	}
 
 	var outputFile *os.File
+	var err error
 	if s.OutputFileName != "" {
-		outputFile, err := os.Create(s.OutputFileName)
```
Comment (philoserf) [inline main.go]: use the created variables rather than shadow them with `:=`

File: main.go [hunk 10]
```diff
@@ -525,19 +514,21 @@ func Process(s *State) {
 	close(resultChan)
 	printerGroup.Wait()
 	if s.OutputFile != nil {
-		outputFile.Close()
+		if err := outputFile.Close(); err != nil {
```
Comment (philoserf) [inline main.go]: check and log errors

File: main.go [hunk 11]
```diff
@@ -572,44 +563,44 @@ func SetupDir(s *State) bool {
...
-	subdomain := word + "." + s.Url
+func processDNSEntry(s *state, word string, resultChan chan<- result) {
+	subdomain := word + "." + s.URL
 	ips, err := net.LookupHost(subdomain)
 
 	if err == nil {
-		if !s.IsWildcard || !s.WildcardIps.ContainsAny(ips) {
-			result := Result{
+		if !s.IsWildcard || !s.WildcardIps.containsAny(ips) {
+			r := result{
```
Comment (philoserf) [inline main.go]: use a short name for the local variable to keep from having too many similar names

**Comment by philoserf** [comment]
This PR is far from perfect.

As I look over GitHub rather than just the code, I see there are a few interesting bits in the milestones. I’ll leave this for discussion and begin work one of the items in milestone 1.4.

**Comment by OJ** [comment]
Apologies for being dark. Hard to find the time to take a breath at the moment!

**Comment by philoserf** [comment]
No issue. No problem. These weren’t significant except as style examples.

**Pull request closed without merging**